### PR TITLE
Create validation.md

### PR DESCRIPTION
**Pull request by satra**
@yarikoptic and @bendichter - i'd like to use this to figure out what we use to validate NWB files for DANDI. consider this the list of things we can enforce. we may have to grandfather in existing dandisets, but moving forward i want to be stricter about collecting information at the user side.

### CODE REVIEW

**Comment by satra** [comment]
@bendichter - also for NWB, perhaps you can list the location of such info in NWB.

File: doc/design/validation.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+- subject id
+- age in iso8601 interval format or datetime of birth
+- age reference 
+- sex
+
+### NWB organization disambiguation requirements
+
+At least one of the following should be present. in case of having to disambiguate
+- session
+- sample
```
Comment (yarikoptic) [inline doc/design/validation.md:16]: I am not sure if there is a generic notion of a "sample" in NWB, so I guess it is for the `tissue_sample_id` [*].

[*] forgot that it is coming probably from `ndx-dandi-icephys` and wasted time to search it up in the main nwb schema, uff

Comment (yarikoptic) [reply doc/design/validation.md:16]: and what about `slice_id`?  might have both I guess (multiple slices from the same sample).

I guess in our metadata schema they both (tissue and slice) would be `BioSample` but of different  `sampleType`, right?

**Review by yarikoptic — COMMENTED** [review]
initial comments/questions

File: doc/design/validation.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Requirements for being a valid DANDI asset
+
+## General
+1. All assets must have a non-zero file size.
+
+## NWB files
+- subject id
```
Comment (bendichter) [inline doc/design/validation.md:7]:
```suggestion
- subject id (nwb: NWBFile.subject.subject_id)
```

File: doc/design/validation.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+- age in iso8601 interval format or datetime of birth
```
Comment (bendichter) [inline doc/design/validation.md:8]:
```suggestion
- age in iso8601 interval format or datetime of birth (nwb.subject.age or nwb.subject.date_of_birth)
```

File: doc/design/validation.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+- age reference 
+- sex
+
+### NWB organization disambiguation requirements
+
+At least one of the following should be present. in case of having to disambiguate
+- session
+- sample
+- cell
```
Comment (bendichter) [inline doc/design/validation.md:17]: we don't have a notion of cell in core either

File: doc/design/validation.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+- probe
```
Comment (bendichter) [inline doc/design/validation.md:18]: what do you mean by this? Probe type, or specific probe serial number?

Comment (satra) [reply doc/design/validation.md:18]: probe identifier - i don't think we found a place to get probe type information but if that exists, let us know.

File: doc/design/validation.md (hunk 1 above)
Comment (satra) [reply doc/design/validation.md:16]: correct.

Comment (yarikoptic) [reply doc/design/validation.md:16]: was that is the answer to the last question, or to both? ;-)

Comment (satra) [reply doc/design/validation.md:16]: the last question (biosample of different sampletype) as implemented currently in the extractor.

File: doc/design/validation.md (hunk 2 above)
Comment (yarikoptic) [reply doc/design/validation.md]: I think @bendichter 's idea to add explicit nwb schema paths is a nice one.  But I see only that @satra got them resolved without comments/changes.   Was that intentional?

File: doc/design/validation.md (hunk 4 above)
Comment (yarikoptic) [reply doc/design/validation.md:17]: is it in `ndx-dandi-icephys` or some other extension?

File: doc/design/validation.md (hunk 2 above)
Comment (satra) [reply doc/design/validation.md]: no i think i exposed a weird github bug. i added them to batch, but something refreshed and i hit refresh and my batch disappeared.

Comment (satra) [inline doc/design/validation.md]:
```suggestion
- subject id (nwb.subject.subject_id)
```

Comment (satra) [reply doc/design/validation.md]: closing this - they have been merged into a commit. so not a bug.

**Comment by yarikoptic** [comment]
moved back to draft since never finalized. I think it was waiting your responses @satra and @bendichter but may be we should reconsider it in full again since there were considerable number of changes to validation since then.

**Comment by yarikoptic** [comment]
it was never followed up on, and I am not sure how much of it is actually reflective of current/desired state, and either it should be here at all, or be a part of some user facing documentation (e.g. handbook) where we outline what we support already. So I will close for now, but if you feel different and plan to finish it up @satra -- feel welcome to reopen.

**Pull request closed without merging**